src/Azure/Orleans.Streaming.EventHubs/Providers/EventDataGeneratorStreamProvider/EventHubPartitionDataGenerator.cs

```diff
@@ -57,18 +57,13 @@ public bool TryReadEvents(int maxCount, out IEnumerable<EventData> events)
                 var eventData = EventHubBatchContainer.ToEventData<int>(this.serializationManager, this.StreamId.Guid, this.StreamId.Namespace,
                     this.GenerateEvent(this.SequenceNumberCounter.Value), RequestContextExtensions.Export(this.serializationManager));
 
-                //set partition key
-                eventData.SetPartitionKey(this.StreamId.Guid.ToString());
+               var wrapper = new WrappedEventData(eventData.Body, eventData.Properties, eventData.SystemProperties,
+                    partitionKey: StreamId.Guid.ToString(),
+                    offset: DateTime.UtcNow.Ticks,
```

Comment:
I'm not sure about this change. Previously offset was set to `StreamId.Guid.ToString() + now.ToString`, but this is impossible now that it's `long`.